apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
I mean that we have these 2 cases anyway, we need to decide if the promise resolution happens at 1 or at 2.